srtcore/core.cpp

```diff
@@ -14165,9 +14242,9 @@ int CUDTGroup::sendBackup(const char *buf, int len, SRT_MSGCTRL& w_mc)
         // no sendable, a new link needs to be activated anyway), and if the
         // priority has a lower number.
         if (sendable_pri.empty()
-                || (!idlers.empty() && idlers[0]->priority < *sendable_pri.begin() ))
+                || (!idlers.empty() && idlers[0]->weight < *sendable_pri.begin() ))
```

Comment:
There are a lot of places where `priority` is renamed to `weight`. This can also be moved to a separate PR to reduce the scope of the review.
- [ ] Move `priority` renaming to a separate PR.

Reply:
Renaming `priority` to `weight` is a part of this PR, just like all changes that introduce new fields for the exclusive use of balancing groups.

The implementation of balancing groups with this field named "priority" doesn't make any sense. Also there were two parts of it in two different PRs, so extracting them would be purely synthetic.